projects/manage-offences-and-delius/src/dev/kotlin/uk/gov/justice/digital/hmpps/data/DataLoader.kt

```diff
@@ -1,25 +1,36 @@
 package uk.gov.justice.digital.hmpps.data
 
 import jakarta.annotation.PostConstruct
+import jakarta.persistence.EntityManager
+import jakarta.transaction.Transactional
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.boot.context.event.ApplicationReadyEvent
 import org.springframework.context.ApplicationListener
 import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.data.generator.DataGenerator
 import uk.gov.justice.digital.hmpps.data.generator.UserGenerator
+import uk.gov.justice.digital.hmpps.repository.DetailedOffenceRepository
+import uk.gov.justice.digital.hmpps.repository.ReferenceDataRepository
 import uk.gov.justice.digital.hmpps.user.AuditUserRepository
 
 @Component
 @ConditionalOnProperty("seed.database")
 class DataLoader(
-    private val auditUserRepository: AuditUserRepository
+    private val auditUserRepository: AuditUserRepository,
+    private val entityManager: EntityManager,
+    private val referenceDataRepository: ReferenceDataRepository,
```

Comment:
Weirdly it didn't work using entityManager for the others.  It throws a:
```
org.hibernate.PersistentObjectException: detached entity passed to persist: uk.gov.justice.digital.hmpps.entity.DetailedOffence
```

which I'm guessing is because we're generating the IDs ourselves?